src/java/com/google/cloud/bigquery/dwhassessment/hooks/logger/EventRecordConstructor.java

```diff
@@ -0,0 +1,218 @@
+    List<String> tableNames = new ArrayList<>();
+    for (Entity entity : entities) {
+      if (entity.getType() == TABLE || entity.getType() == PARTITION) {
+        tableNames.add(entity.getTable().getDbName() + "." + entity.getTable().getTableName());
+      }
+    }
+    return tableNames;
+  }
+
+  private static String getUser(HookContext hookContext) {
+    return hookContext.getUgi().getShortUserName();
+  }
+
+  private static String getRequestUser(HookContext hookContext) {
+    String requestUser = hookContext.getUserName();
+    return requestUser == null ? hookContext.getUgi().getUserName() : requestUser;
+  }
+
+  private static ExecutionMode getExecutionMode(List<ExecDriver> mrTasks, List<TezTask> tezTasks) {
+    if (tezTasks.size() > 0) {
+      // Need to go in and check if any of the tasks is running in LLAP mode.
+      for (TezTask tezTask : tezTasks) {
+        if (tezTask.getWork().getLlapMode()) {
+          return ExecutionMode.LLAP;
+        }
+      }
+      return ExecutionMode.TEZ;
+    } else if (mrTasks.size() > 0) {
+      return ExecutionMode.MR;
+    } else {
+      return ExecutionMode.NONE;
+    }
+  }
+
+  private static String getQueueName(ExecutionMode mode, HiveConf conf) {
+    switch (mode) {
+      case LLAP:
+        return conf.get(HiveConf.ConfVars.LLAP_DAEMON_QUEUE_NAME.varname);
+      case MR:
+        return conf.get(MR_QUEUE_NAME.getConfName());
+      case TEZ:
+        return conf.get(TEZ_QUEUE_NAME.getConfName());
+      case NONE:
+      default:
+        return null;
+    }
+  }
+
+  private static String getHiveInstanceAddress(HookContext hookContext) {
+    String hiveInstanceAddress = hookContext.getHiveInstanceAddress();
+    if (hiveInstanceAddress == null) {
+      try {
+        hiveInstanceAddress = InetAddress.getLocalHost().getHostAddress();
+      } catch (UnknownHostException e) {
+        LOG.error("Error trying to get localhost address: ", e);
+      }
+    }
+    return hiveInstanceAddress;
+  }
+
+  private static String getHiveInstanceType(HookContext hookContext) {
+    return hookContext.isHiveServerQuery() ? "HS2" : "CLI";
+  }
+
+  private static String determineLlapId(HiveConf conf, ExecutionMode mode) {
+    // Note: for now, LLAP is only supported in Tez tasks. Will never come to MR; others may
+    // be added here, although this is only necessary to have extra debug information.
+    if (mode == ExecutionMode.LLAP) {
+      // In HS2, the client should have been cached already for the common case.
+      // Otherwise, this may actually introduce delay to compilation for the first query.
+      String hosts = HiveConf.getVar(conf, HiveConf.ConfVars.LLAP_DAEMON_SERVICE_HOSTS);
+      if (hosts != null && !hosts.isEmpty()) {
+        try {
+          return LlapRegistryService.getClient(conf).getApplicationId().toString();
+        } catch (IOException e) {
+          LOG.error("Error trying to get llap instance: ", e);
```

Comment:
1. I would suggest to remove the colon `:` at the end of the error message.
2. It might be helpful to log the contents of `hosts` variable.

Reply:
Done